src/settings.js

```diff
@@ -1776,6 +1776,13 @@ var EVAL_CTORS = 0;
 // [link]
 var TEXTDECODER = 1;
 
+// The default value of 1 means the generated code will use TextEncoder if
+// available and fall back to custom encoding code when it is not available.  
+// If set to 2, we assume TextEncoder is always present and usable, and no
+// fallback JS code will be emitted.
+// [link]
+var TEXTENCODER = 1;
```

Comment:
I'm wondering if we should actually just make this an internal setting?   (i.e. move it to `settings_internal.js`).

Since we don't support targeting browsers that don't support TestEncoder at all (See https://caniuse.com/textencoder and https://github.com/emscripten-core/emscripten/blob/cdec9c1d60a760abb4ea88bcbce10948a54baeff/tools/feature_matrix.py#L18-L30).

That way we can just default to `=2` when ENVIRONMENT is web or node and `=1` when we need shell support?

The cost of adding more public options is pretty high (IMHO) so if we can avoid it, even initially, that would be great!